src/webglimpse/plot/axis.ts

```diff
@@ -83,14 +109,27 @@ export class Axis1D {
     }
 
     pan(vAmount: number) {
+        if (vAmount + this._vMin < this._vMinLimit) {
+            this._vMax = this._vMinLimit + this.vSize;
+            this._vMin = this._vMinLimit;
+            this._limitsChanged.fire();
+            return;
+        } else if (vAmount + this._vMax > this._vMaxLimit) {
+            this._vMin = this._vMaxLimit - this.vSize;
+            this._vMax = this._vMaxLimit;
+            this._limitsChanged.fire();
+            return;
+        }
         this._vMin += vAmount;
```

Comment:
Just put this into an else and remove the duplicate _limitsChanged.fire() calls and the returns.